src/renderer/services/dex.js

```diff
@@ -590,16 +590,22 @@ export default {
 
     if (order.price) {
       // limit order
+      let depositMakerQuantity = false;
+
       if (sellAssetHolding.canPull === false && order.quantity.isGreaterThan(0)) {
         // this is an MCT based token that can not be pulled from our DEX contract, have to send a deposit first
-        totalQuantityToSell = order.side === 'Buy' ? order.quantity.multipliedBy(order.price) : order.quantity;
+        depositMakerQuantity = true;
       } else if (sellAssetHolding.decimals < 8) {
         // this is a token with < 8 decimals, NEO for example, make the deposit of the minimum amount needed to make the order
-        totalQuantityToSell = order.side === 'Buy' ? order.quantity.multipliedBy(order.price) : order.quantity;
+        depositMakerQuantity = true;
+      } else if (order.offersToTake.length > 0 && order.quantityToMake.isGreaterThan(0)) {
+        // we have maker and taker quantities, need to deposit the maker quanity first because we don't know the order they will be confirmed
```

Comment:
I think deposit would be necessary regardless of the order they are confirmed. Maybe update the comment.

Reply:
If it confirmed the takers first then the makers would pull the required tokens in during addOffer.

Reply:
ah yes, thanks for clarifying